### DISCUSSION TITLE
K1 Max camera stops

### DISCUSSION THREAD

**Discussion by flippnswithez** [Guilouz/Creality-Helper-Script-Wiki#508] [Issues]
Everything worked well until this last update. Now the camera starts when the machine is turned on. While printing the camera stops working and says unable to load Webcam. I have to reboot the printer to get it to work again.

**Comment by CyberCowboy** [comment]
Also getting this problem on a K1 with the self-added camera.

**Comment by MateuszKrawczuk** [comment]
I have same issue with K1C.

**Reply by MateuszKrawczuk** [reply]
I notice this error in dmesg:
```
[ 2393.669173] helix-venc 13200000.helix: h264 start streaming 
[ 2437.675124] usb 1-1.3: USB disconnect, device number 29
[ 2437.682639] dwc2 13500000.otg_new: dwc2_hc_halt() Channel can't be halted
[ 2437.728049] helix-venc 13200000.helix: h264 stop streaming !
[ 2437.742544] helix-venc 13200000.helix: [1] encoder release
[ 2437.756261] helix-venc 13200000.helix: h264 stop streaming !
[ 2437.769781] helix-venc 13200000.helix: [2] encoder release
[ 2437.774853] auto_uvc.sh (22031): drop_caches: 3
[ 2437.780507] helix-venc 13200000.helix: jpgd stop streaming !
[ 2437.786509] helix-venc 13200000.helix: [0] encoder release
[ 2437.895737] ------------[ cut here ]------------
[ 2437.901658] WARNING: CPU: 1 PID: 21071 at fs/sysfs/group.c:237 sysfs_remove_group+0x60/0xbc()
[ 2437.911026] sysfs group 8080f570 not found for kobject 'media0'
[ 2437.917380] Modules linked in: x2000_icodec_sound_card(O) tlsc6x(O) soc_watchdog(O) soc_mac_driver(O) soc_efuse(O) lcd_st7701_9bit_spi_tjc_480_800(O) hci_uart_h5_kernel_4_4_94(O) gt9xx_touch(O) cywdhd(O) soc_dtrng(O) soc_msc(O) soc_icodec(O) soc_aic(O) soc_audio_dma(O) soc_fb(O) soc_fb_layer_mixer(O) soc_rotator(O) pwm_backlight(O) soc_pwm(O) soc_gpio(O) soc_i2c(O) soc_utils(O) rmem_manager(O) utils(O)
[ 2437.954968] CPU: 1 PID: 21071 Comm: cam_app Tainted: G        W  O    4.4.94 #342
[ 2437.962692] Stack : 80877a4a 00000045 00000000 80880000 00000000 00000000 8007a034 807f0a47
[ 2437.962692] 	  80742300 00000001 0000524f 80874768 807e0000 807f0000 00000009 8007a028
[ 2437.962692] 	  00000009 8027aef0 00000001 00000000 8074787c 88785b4c 88785b30 800ac154
[ 2437.962692] 	  8015110c 8003976c 807e0000 807f0000 88785b4c 00000002 8adf16c0 8c13bd50
[ 2437.962692] 	  00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
[ 2437.962692] 	  ...
[ 2437.999514] Call Trace:
[ 2438.002051] [<8002250c>] show_stack+0x70/0x8c
[ 2438.006560] [<80270524>] dump_stack+0x94/0xd0
[ 2438.011068] [<8003991c>] warn_slowpath_common+0xa0/0xd0
[ 2438.016462] [<80039978>] warn_slowpath_fmt+0x2c/0x38
[ 2438.021596] [<8015110c>] sysfs_remove_group+0x60/0xbc
[ 2438.026812] [<802de6a0>] device_del+0x44/0x1f0
[ 2438.031399] [<802de860>] device_unregister+0x14/0x28
[ 2438.036534] [<803b9d14>] uvc_delete+0x68/0x12c
[ 2438.041128] [<80399f84>] v4l2_device_release+0x120/0x16c
[ 2438.046622] [<802dd908>] device_release+0x3c/0xb4
[ 2438.051485] [<80272d48>] kobject_put+0xd0/0x10c
[ 2438.056172] [<8039a0d8>] v4l2_release+0x6c/0x84
[ 2438.060855] [<800ed694>] __fput+0xfc/0x1e8
[ 2438.065093] [<80053488>] task_work_run+0xac/0xc4
[ 2438.069870] [<8003b508>] do_exit+0x3ac/0x814
[ 2438.074285] [<8003c780>] do_group_exit+0x58/0xb4
[ 2438.079061] [<80047454>] get_signal+0x540/0x548
[ 2438.083748] [<8002017c>] do_signal+0x14/0x1a0
[ 2438.088253] [<80021318>] do_notify_resume+0x50/0xc8
[ 2438.093294] [<8001c280>] work_notifysig+0x10/0x18
[ 2438.098154] 
[ 2438.109004] ---[ end trace 46fe0a5e4983f212 ]---
[ 2440.621779] usb 1-1.3: new high-speed USB device number 30 using dwc2
[ 2440.801778] usb 1-1.3: New USB device found, idVendor=a108, idProduct=2231
[ 2440.810270] usb 1-1.3: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 2440.817902] usb 1-1.3: Product: CCX2F3299
[ 2440.822458] usb 1-1.3: Manufacturer: UnionImage Co.,Ltd
[ 2440.827980] usb 1-1.3: SerialNumber: 1234567890
[ 2440.843938] uvcvideo: Found UVC 1.00 device CCX2F3299 (a108:2231)
[ 2440.972377] auto_uvc.sh (22100): drop_caches: 3
[ 2441.084186] dwc2 13500000.otg_new: dwc2_hc_halt() Channel can't be halted
[ 2441.513657] helix-venc 13200000.helix: V4L2_CID_JPEG_COMPRESSION_QUALITY = 1
[ 2441.521146] helix-venc 13200000.helix: Create instance [0]@88218000 m2m_ctx=849ef800
[ 2441.529370] helix-venc 13200000.helix: 13200000.helix vcodec [0]
[ 2441.536242] helix-venc 13200000.helix: jpgd start streaming 
[ 2441.542292] helix-venc 13200000.helix: V4L2_CID_JPEG_COMPRESSION_QUALITY = 1
[ 2441.549899] helix-venc 13200000.helix: Create instance [1]@8821c000 m2m_ctx=849ec800
[ 2441.558266] helix-venc 13200000.helix: 13200000.helix vcodec [1]
[ 2441.572581] rc_mode: vbr
[ 2441.575553] helix-venc 13200000.helix: h264 start streaming 
[ 2441.581843] helix-venc 13200000.helix: V4L2_CID_JPEG_COMPRESSION_QUALITY = 1
[ 2441.589298] helix-venc 13200000.helix: Create instance [2]@8821e000 m2m_ctx=849eb800
[ 2441.597937] helix-venc 13200000.helix: 13200000.helix vcodec [2]
[ 2441.605590] rc_mode: vbr
[ 2441.609055] helix-venc 13200000.helix: h264 start streaming 
[ 2453.095383] usb 1-1.3: USB disconnect, device number 30
[ 2453.102634] dwc2 13500000.otg_new: dwc2_hc_halt() Channel can't be halted
[ 2453.139765] helix-venc 13200000.helix: h264 stop streaming !
[ 2453.146001] helix-venc 13200000.helix: [1] encoder release
[ 2453.152065] helix-venc 13200000.helix: h264 stop streaming !
[ 2453.158046] helix-venc 13200000.helix: [2] encoder release
[ 2453.164285] helix-venc 13200000.helix: jpgd stop streaming !
[ 2453.177336] helix-venc 13200000.helix: [0] encoder release
[ 2453.197215] auto_uvc.sh (22394): drop_caches: 3
[ 2453.298537] ------------[ cut here ]------------
[ 2453.303451] WARNING: CPU: 0 PID: 22141 at fs/sysfs/group.c:237 sysfs_remove_group+0x60/0xbc()
[ 2453.312411] sysfs group 8080f570 not found for kobject 'media0'
[ 2453.318816] Modules linked in: x2000_icodec_sound_card(O) tlsc6x(O) soc_watchdog(O) soc_mac_driver(O) soc_efuse(O) lcd_st7701_9bit_spi_tjc_480_800(O) hci_uart_h5_kernel_4_4_94(O) gt9xx_touch(O) cywdhd(O) soc_dtrng(O) soc_msc(O) soc_icodec(O) soc_aic(O) soc_audio_dma(O) soc_fb(O) soc_fb_layer_mixer(O) soc_rotator(O) pwm_backlight(O) soc_pwm(O) soc_gpio(O) soc_i2c(O) soc_utils(O) rmem_manager(O) utils(O)
[ 2453.356667] CPU: 0 PID: 22141 Comm: cam_app Tainted: G        W  O    4.4.94 #342
[ 2453.364391] Stack : 80877a4a 00000045 00000000 80880000 00000000 00000000 8007a034 807f0a47
[ 2453.364391] 	  80742300 00000000 0000567d 80874768 807e0000 807f0000 00000009 8007a028
[ 2453.364391] 	  00000009 8027aef0 00000001 00000000 8074787c 89579b4c 89579b30 800ac154
[ 2453.364391] 	  8015110c 8003976c 807e0000 807f0000 89579b4c 00000002 8c5c6340 8c13e630
[ 2453.364391] 	  00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
[ 2453.364391] 	  ...
[ 2453.401214] Call Trace:
[ 2453.403751] [<8002250c>] show_stack+0x70/0x8c
[ 2453.408261] [<80270524>] dump_stack+0x94/0xd0
[ 2453.412768] [<8003991c>] warn_slowpath_common+0xa0/0xd0
[ 2453.418175] [<80039978>] warn_slowpath_fmt+0x2c/0x38
[ 2453.423317] [<8015110c>] sysfs_remove_group+0x60/0xbc
[ 2453.428545] [<802de6a0>] device_del+0x44/0x1f0
[ 2453.433141] [<802de860>] device_unregister+0x14/0x28
[ 2453.438280] [<803b9d14>] uvc_delete+0x68/0x12c
[ 2453.442878] [<80399f84>] v4l2_device_release+0x120/0x16c
[ 2453.448369] [<802dd908>] device_release+0x3c/0xb4
[ 2453.453230] [<80272d48>] kobject_put+0xd0/0x10c
[ 2453.457916] [<8039a0d8>] v4l2_release+0x6c/0x84
[ 2453.462599] [<800ed694>] __fput+0xfc/0x1e8
[ 2453.466837] [<80053488>] task_work_run+0xac/0xc4
[ 2453.471611] [<8003b508>] do_exit+0x3ac/0x814
[ 2453.476025] [<8003c780>] do_group_exit+0x58/0xb4
[ 2453.480797] [<80047454>] get_signal+0x540/0x548
[ 2453.485483] [<8002017c>] do_signal+0x14/0x1a0
[ 2453.489986] [<80021318>] do_notify_resume+0x50/0xc8
[ 2453.495029] [<8001c280>] work_notifysig+0x10/0x18
[ 2453.499887] 
[ 2453.506050] ---[ end trace 46fe0a5e4983f213 ]---
[ 2456.162029] usb 1-1.3: new high-speed USB device number 31 using dwc2
[ 2456.342083] usb 1-1.3: New USB device found, idVendor=a108, idProduct=2231
[ 2456.352585] usb 1-1.3: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 2456.366555] usb 1-1.3: Product: CCX2F3299
[ 2456.371566] usb 1-1.3: Manufacturer: UnionImage Co.,Ltd
[ 2456.377520] usb 1-1.3: SerialNumber: 1234567890
[ 2456.394296] uvcvideo: Found UVC 1.00 device CCX2F3299 (a108:2231)
```

**Reply by Muratore73** [reply]
My K1C has the same log-entries. Did you ever get rid of these usb-disconnects?

**Comment by AcBCDN** [comment]
My camera has stopped working as well since the last update. The CPU maxing out at 100%.

**Comment by TheWebMachine** [comment]
Are you guys by chance heating your enclosed chamber?

I have had zero problems with the K1C camera on the latest script and firmware...until I closed it up, turned on my chamber heater (my own added that tops out at 50°C), and attempted to print ABS. 

Twice now, the camera has cut out once the temps get high in the chamber. My chamber temps top out around 70°C during the print - 60°C reported by the cheap-ass temp sensor, but 70°C confirmed by me with an IR gun. The camera, of course, gets to this temp, too. I'm wondering if the [certainly cheap, based on vid quality,] camera sensor they used has an upper temp limit of a wee bit lower than that and it is simply cutting out and crashing the driver.

I do not have any cam issues when printing PLA with the top off, as the camera never gets above 40°C. Unfortunately, I seem to remember the same thing happening on 1.3.3.5, but I had only done one ASA print at that point and just rebooted when it was done and moved on with my usual PLA/PETG prints with the cover off and chamber unheated.

So, are y'all closing up the chamber during print and/or heating it when your cam dies? Have you checked the temp of the cam housing after it dies?

**Reply by TheWebMachine** [reply]
I spent the night running multiple prints, even the same ABS print repeatedly with both closed/heated and open chambers. I'm certain that heat is what is killing the camera, at least in my case. As soon as temps in the chamber get above about 60°C, the cam cuts out. Even without the added chamber heater, this can happen on a long enough ABS/ASA print.

My next steps are:
1. Full teardown of the camera and identify all component model numbers within
2. Perform thermal stress testing directly on specific components to identify the failing components and determine if something simple, like a cheap capacitor, can be swapped
3. Determine if a passive or active (unlikely) cooling solution can be employed; tiny heatsinks are easy to come by
4. Identify alternative camera modules that might provide drop-in replacement but provide higher heat tolerance

Another potential avenue I could use some help with:
-  Identify if the cam issue is merely causing occasional glitches that could potentially be recovered from via software after the driver crashes, or if the cam is totally dead until cooldown

**Comment by AcBCDN** [comment]
Software or heat the time the camera stops working  is between 1--1.5 hrs consistently.  Chamber was 34C at the point of failure. IT also appears the mjpg streamer is custom(??).

-----------------
tail /var/log/messages 
Apr 10 13:06:24 K1Max-5780 user.err kernel: [ 5656.719817] usb 1-1: clear tt 1 (9263) error -145
Apr 10 13:06:24 K1Max-5780 user.info kernel: [ 5657.051643] usb 1-1.1: USB disconnect, device number 38
Apr 10 13:06:24 K1Max-5780 user.info kernel: [ 5657.263755] usb 1-1.1: new full-speed USB device number 39 using dwc2
Apr 10 13:06:24 K1Max-5780 user.info kernel: [ 5657.357227] usb 1-1.1: New USB device found, idVendor=1a86, idProduct=55d3
Apr 10 13:06:24 K1Max-5780 user.info kernel: [ 5657.364432] usb 1-1.1: New USB device strings: Mfr=0, Product=2, SerialNumber=3
Apr 10 13:06:24 K1Max-5780 user.info kernel: [ 5657.372244] usb 1-1.1: Product: USB Single Serial
Apr 10 13:06:24 K1Max-5780 user.info kernel: [ 5657.377417] usb 1-1.1: SerialNumber: 5687025049
Apr 10 13:06:24 K1Max-5780 user.info kernel: [ 5657.383068] cdc_acm 1-1.1:1.0: ttyACM0: USB ACM device
Apr 10 13:06:37 K1Max-5780 authpriv.info dropbear[20495]: Child connection from 192.168.1.94:60380
Apr 10 13:06:39 K1Max-5780 authpriv.notice dropbear[20495]: Password auth succeeded for 'root' from 192.168.1.94:60380
root@K1Max-5780 /root [#] lsusb
Bus 001 Device 004: ID 0c45:6366 Microdia 
Bus 001 Device 003: ID 23a5:5786  
Bus 001 Device 039: ID 1a86:55d3 QinHeng Electronics USB2.0 Hub
Bus 001 Device 002: ID 05e3:0610 Genesys Logic, Inc. 4-port hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub

**Reply by TheWebMachine** [reply]
Interesting. Can you check the temp of the camera's housing at the time of failure? I am finding the chip inside these cams - I'll be posting a write-up on that chip soon - gets to 55-60°C in ambient temps (34-35°C) all on its own just steaming video. There is no heatsink on that chip and woefully inadequate ventilation in the case. I suspect the heated bed's proximity to the camera may be, in some cases, enough to kill the cam even if not enclosed. I'm still testing different scenarios. I, too, am seeing that it takes a bit of time into the print before the cam fails, but the cam and timelapse snapshots are working fine up until the moment of failure, even if the cam was working fine at idle for hours beforehand.

Anyway, I've already begun modding my camera to try to bring that chip's temps down to see if that stops the failures. As soon as I have some conclusions on that front, I will be posting my findings here. I really do think heat may be to blame here. My preliminary tests - adding a heatsink to the chip and modifying the back panel of the cam to allow airflow - are showing promise so far, but the hours of waiting between failure points is slowing me down. haha

Teaser: the chip - in the K1C included camera, anyway - is the Ingenic T31ZL, though I suspect Creality is using the exact same board in the K1[ Max] add-on cameras, too. The chip is capable of way more than 720@15fps, but I'm pretty sure Creality "underclocked" and capped to 720@15fps precisely for thermal reasons. My investigation continues...

**Reply by TheWebMachine** [reply]
Oh, and I think that hub may be built onto the camera board to provide access to several interfaces in the camera over a single USB connection to the board. The chip is a dual-core DDR2 bit of kit with a bunch of things added in like AI defogging and such. 
![Ingenic-T31-MIPS-RISC-V-Video-Processor](https://github.com/Guilouz/Creality-K1-Series/assets/1045052/5d14623d-4725-49de-a487-9b2b91dc838e)

**Reply by AcBCDN** [reply]
Failed: 53 mins into a 4 hr PLA print with lid removed.  Chamber was 34C the Camera housing was 31C . PLA 220C Bed 50C.

**Reply by TheWebMachine** [reply]
Thanks for the info!

**Reply by TheWebMachine** [reply]
How willing are you to pull the back cover off that camera and snap a pic of the board? 😁 Just want to confirm we are talking about the same module despite different printers. I only need a reasonably clear shot of the back side with the chips.

**Reply by AcBCDN** [reply]
Have never looked. If its not too involved I will take a look after the print is finished.

**Reply by TheWebMachine** [reply]
If it's similar to the K1C, the camera case is being held together by a couple of basic, albeit small, (+) screws and easily comes apart. Just be careful because the back cover is probably also holding the camera board assembly in the case; no other notches or glue keeping it in there. I dropped my board the first time taking it apart because I had it tilted and it just comes loose from the front half when you take the cover off. 🤣

**Reply by AcBCDN** [reply]
The images of the CAM
![back](https://github.com/Guilouz/Creality-K1-Series/assets/6889303/14b15556-31db-4987-bb82-bc991b0103b5)
![front](https://github.com/Guilouz/Creality-K1-Series/assets/6889303/5bb60818-66b1-4551-9036-94a802723b7a)

**Reply by TheWebMachine** [reply]
Damn, I'm gonna have to leave you guys to your own issue. Totally different chip, which means we are likely not having the same problem, whether it is hardware or software/drivers to blame....unless it turns out to be something in software like the mpegstreamer (which means I'm chasing a dead end, anyway). 🤷🏻‍♂️

**Comment by flippnswithez** [comment]
My issue actually has resolved itself🤷 I have only been printing ASA with the machine closed up.

**Reply by TheWebMachine** [reply]
Total side topic: Which brand of ASA are you using? I am finding Polymaker's ASA to be a nightmare to tune settings for.

**Reply by flippnswithez** [reply]
iSANGHU and ELEGOO from Amazon at the moment. iSANGHU prints came out great with no problems at all, I tried their red. The Elegoo is black and I had to do a little tuning to get everything good.  I also randomly tried a gold ABS from MG Chemicals. I was surprised at how great it was. My only complaint with MG Chemicals is they filled the roll so much that my first couple of prints failed because the filament kept jumping off the roll and getting tangled.

**Reply by TheWebMachine** [reply]
Ah, this is why I prefer spool roller stands over center spindle holders. Go print the Prusa Mini+ spool holder from here (plus obtain the prescribed bearings) and spool jump tangles will become a thing of the past:

https://www.printables.com/model/57214-mini-printable-parts/files

If you also print with +KG spools (2kg wide, etc), print a pair of larger adjustable rails from here:

https://www.printables.com/model/89695-prusa-spool-holder-extensions

**Comment by firepower9966** [comment]
My camera died and creality replaced it. these are photos of dead camera from my K1 Max. 


![IMG_20240413_120506](https://github.com/Guilouz/Creality-K1-Series/assets/3244372/a4af9c1f-e222-49de-bcc8-25c9b5dd5375)
![IMG_20240413_120418](https://github.com/Guilouz/Creality-K1-Series/assets/3244372/527c5b18-4021-4afc-907b-811abe89343b)

**Reply by TheWebMachine** [reply]
This is the same cam board that comes with the k1c. I'm still testing from the "thermal crash" angle. Should have some more info this weekend. I wonder if your new camera is the same or the other one pictured above. Maybe Creality is realizing a mistake was made in choosing the T31ZL? 🤷🏻‍♂️

**Comment by firepower9966** [comment]
I wired it to USB cable, 5V drops to 4.6V, no 3V3 and it gets very hot. is not recognised as USB device on PC.

**Comment by firepower9966** [comment]
more info here with data sheets of parts used. 
https://discord.com/channels/1154500511777693819/1228572148541558884

**Reply by kozross** [reply]
Could you copy-and-paste this over? I can't access that channel.

**Reply by B1T3X** [reply]
Yeah I'd be interested in that too :)

**Comment by jattie-ire** [comment]
> Everything worked well until this last update. Now the camera starts when the machine is turned on. While printing the camera stops working and says unable to load Webcam. I have to reboot the printer to get it to work again.

After some digging I found this file and command on the K1 Max:

`ACTION=reload /usr/bin/auto_uvc.sh`

This reloads the video services without a reboot.

**Reply by Batfinks** [reply]
I have the same problem.  I have only been printing ABS with the chamber closed and the camera dies probably less than 15 min into the print.  I tried your reload in an ssh during a print, did not do anything.  It looks like the command was accepted, but no change in FLUIDD.

**Reply by jattie-ire** [reply]
It unfortunately does not work in all failure modes. Check your dmesg logs. Do you get repeat connection disconnection messages for usb devices? If so then you probably have the same issue I had. 

You can also run lsusb command and look at the device numbers. If you have a number bigger than 4 you probably have a hardware problem.

**Comment by iluminari3d** [comment]
> Everything worked well until this last update. Now the camera starts when the machine is turned on. While printing the camera stops working and says unable to load Webcam. I have to reboot the printer to get it to work again.

Hi guys this problem has been driving me crazy for 4 months. I bought two M1 Max at the beginning of the year. After a couple of months of use on one of the two this problem starts. 

First of all I replaced the camera, but the problem was not there. A few weeks ago Creality released this FAQ:

https://wiki.creality.com/en/k1-flagship-series/k1-max/troubleshooting/unable-to-detect-camera

Trying to disconnect the lidar the camera problem went away but I had no lidar calibration.Thinking it was the problem I tried to replace it with the "working" one from the other printer. The problem is to buy again. I only have two alternatives, change the board at the head of the print head or wait for a firmware update.

I bought this one here which is often on sale for around $20:
https://a.aliexpress.com/_EyMbBmg

It arrived a few days ago and the problem seems to have finally disappeared. I made a few prints, longer than the time it used to take to disconnect. Everything seems to be working!

I hope this was helpful and that you can solve the problem. Creality support with me unfortunately was non-existent! Happy printing❤️

**Reply by CyberCowboy** [reply]
Unfortunately I have the OG k1 so no lidar to disconnect.  Glad it's working for you though.

**Comment by NexGen-3D-Printing** [comment]
I just installed a chamber heater in my K1 Max and the camera is now locking up, so it has the same issue as K1C, if anyone stumbles across this thread, this is the fix: https://www.printables.com/model/904768-nexgen3d-creality-k1c-camera-fix

**Comment by YEET1201** [comment]
If the camera suddenly turns off and never re connects even after restarting the printer does that mean its confirmed broken?

**Reply by jattie-ire** [reply]
There is a small chance that it could be a cable connection issue but it is very likely that the camera board failed. The electronics are exposed to the heat cycles inside the chamber.

**Comment by NexGen-3D-Printing** [comment]
Correct, the heat produced in this machine will cause the camera to lockup all the time, and I could definately see it dying from this abuse, mine after all this time with a chamber heater going and printing high temp materials, even with blankets and towels thrown over it at times, the camera still works perfectly using my fan cooling mod, I have gotten it hot enought to lockup the toolhead control board and even over heat the steppers, but not the camera :)
